src/pytorch_lightning/serve/servable_module_validator.py

```diff
@@ -0,0 +1,142 @@
+        if not uvicorn_installed:
+            raise ModuleNotFoundError(uvicorn_installed.message)
+
+        # TODO: Add support for the other options
+        if optimization is not None:
+            raise NotImplementedError(f"The optimization {optimization} is currently not supported.")
+        # TODO: Add support for testing with those server services
+        if server != "fastapi":
+            raise NotImplementedError("Only the fastapi server is currently supported.")
+
+        self.optimization = optimization
+        self.host = host
+        self.port = port
+        self.server = server
+        self.resp: Optional[requests.Response] = None
+
+    @rank_zero_only
+    def on_train_start(self, trainer: "pl.Trainer", servable_module: "pl.LightningModule") -> None:
+        if isinstance(trainer.strategy, _NOT_SUPPORTED_STRATEGIES):
+            raise Exception(
+                f"The current strategy {trainer.strategy} used "
+                "by the trainer isn't supported for sanity serving yet."
+            )
+
+        if not isinstance(servable_module, ServableModule):
+            raise TypeError(f"The provided model should be subclass of {ServableModule.__qualname__}.")
+
+        if not is_overridden("configure_payload", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_payload` method needs to be overridden.")
+        if not is_overridden("configure_serialization", servable_module, ServableModule):
+            raise NotImplementedError("The `configure_serialization` method needs to be overridden.")
+        if not is_overridden("serve_step", servable_module, ServableModule):
+            raise NotImplementedError("The `serve_step` method needs to be overridden.")
+
+        process = Process(target=self._start_server, args=(servable_module, self.host, self.port, self.optimization))
+        process.start()
+
+        ready = False
+        while not ready:
+            try:
+                resp = requests.get(f"http://{self.host}:{self.port}/ping")
+                ready = resp.status_code == 200
+            except requests.exceptions.ConnectionError:
+                pass
+            time.sleep(0.1)
+
+        payload = servable_module.configure_payload()
+
+        if "body" not in payload:
+            raise Exception(f"Your provided payload {payload} should have a field body.")
+
+        self.resp = requests.post(f"http://{self.host}:{self.port}/serve", json=payload)
+        process.kill()
+
+    @property
+    def successful(self) -> bool:
+        """Returns whether the model was successfully served."""
+        return self.resp.status_code == 200 if self.resp else False
+
+    def state_dict(self) -> Dict[str, Any]:
+        return {"successful": self.successful, "optimization": self.optimization, "server": self.server}
+
+    @staticmethod
+    def _start_server(servable_model: ServableModule, host: str, port: int, _: bool) -> None:
```

Comment:
did you test this with ddp-spawn?

I'm asking because you do have some locally defined functions here which may be an issue when pickling the callback

Reply:
The function shouldn't be a problem as the callback can be pickled without issues. The local functions aren't deepcopied.

Reply:
Resolved this bug.